ui/main/src/app/services/lightcards.service.ts

```diff
@@ -92,24 +97,39 @@ export class LightCardsService {
         return this.store.pipe(
             select(feedSelectors.selectFeed),
             sample(interval(1000)),
-            filter(() => ((new Date().valueOf()) - this.lastDebounce) > 1000), // we only need to get cards if no debounce arise in 1 seconds) 
-            tap(()=> this.loadingInProgress.next(true)) 
+            filter(() => ((new Date().valueOf()) - this.timeOfLastDebounce) > 1000), // we only need to get cards if no debounce arise in 1 seconds) 
         );
     }
 
     private getLightCardDebounce(): Observable<any> {
         return this.store.pipe(
             select(feedSelectors.selectFeed),
+            tap(() => this.timeOfLastCardReception = new Date().valueOf()),
             debounceTime(200),
-            tap(() => this.lastDebounce = (new Date()).valueOf()),
-            tap(()=> this.loadingInProgress.next(false)) 
+            tap(() => this.timeOfLastDebounce = (new Date()).valueOf())
         );
     }
 
     public getLoadingInProgress() {
         return this.loadingInProgress.asObservable();
     }
 
+    // check every 500ms if we are in a loading card process
+    // we consider we are in a loading card process if we have 3 following times cards not processed by getLightCardDebounce();
+    private checkForLoadingInProgress()
+    {
+        if (this.timeOfLastCardReception > this.timeOfLastDebounce) // at least one card has been received but not yet process by getLightCardDebounce() , it is in progress
+                  {
+                    this.numberOfFollowingTimeDebounceHasNotBeenDone++;
+                    if (this.numberOfFollowingTimeDebounceHasNotBeenDone > 2) this.loadingInProgress.next(true);
+                  }
+        else  {
+            this.loadingInProgress.next(false);
+            this.numberOfFollowingTimeDebounceHasNotBeenDone = 0;
+        }
+        setTimeout(() => this.checkForLoadingInProgress(), 500);
```

Comment:
Defining the various milliseconds durations or card numbers as constants might also help?

Reply:
yes on principle but in this case , i think it is obvious